packages/integrations/cloudflare/src/server.directory.ts

```diff
@@ -38,17 +29,35 @@ export function createExports(manifest: SSRManifest) {
 				Symbol.for('astro.clientAddress'),
 				request.headers.get('cf-connecting-ip')
 			);
+
+			// @deprecated: getRuntime() can be removed, use `Astro.locals.runtime` instead
 			Reflect.set(request, Symbol.for('runtime'), {
-				...runtimeEnv,
+				...context,
 				waitUntil: (promise: Promise<any>) => {
-					runtimeEnv.waitUntil(promise);
+					context.waitUntil(promise);
 				},
 				name: 'cloudflare',
 				next,
 				caches,
 				cf: request.cf,
 			});
-			let response = await app.render(request, routeData);
+
+			let response = await app.render(request, routeData, {
+				env: env,
+				cf: request.cf,
+				runtime: {
+					// request: Request; // not needed because we have Astro.request, thus they are minor differences
+					// functionPath: string; // not needed
+					waitUntil: (promise: Promise<any>) => { context.waitUntil(promise); },
+					// passThroughOnException: () => void; // probably not needed ??
+					// next: (input?: Request | string, init?: RequestInit) => Promise<Response>; // probably not needed ??
```

Comment:
Should we include them anyways, I can't think of a use-case but other users might have some.

Reply:
Usually it's best to pass a tiny set of APIs, and then include new ones on the go, if users need them.